app/src/main/java/com/techcourse/dao/UserDao.java

```diff
@@ -30,14 +31,16 @@ public List<User> findAll() {
         return jdbcTemplate.query(sql, USER_ROW_MAPPER);
     }
 
-    public User findById(final Long id) {
+    public Optional<User> findById(final Long id) {
         final var sql = "select id, account, password, email from users where id = ?";
-        return jdbcTemplate.queryForObject(sql, USER_ROW_MAPPER, id);
+        final User user = jdbcTemplate.queryForObject(sql, USER_ROW_MAPPER, id);
+        return Optional.of(user);
     }
 
-    public User findByAccount(final String account) {
+    public Optional<User> findByAccount(final String account) {
         final var sql = "select id, account, password, email from users where account = ?";
-        return jdbcTemplate.queryForObject(sql, USER_ROW_MAPPER, account);
+        final User user = jdbcTemplate.queryForObject(sql, USER_ROW_MAPPER, account);
+        return Optional.of(user);
     }
```

Comment:
이렇게 하면 NPE 떠요. 다음 단계에서 참고 바랍니다.

```suggestion
        final User user = jdbcTemplate.queryForObject(sql, USER_ROW_MAPPER, account);
        if (user == null) {
            return Optional.empty();
        }
        return Optional.of(user);
} 
```

<img width="360" alt="image" src="https://user-images.githubusercontent.com/73531614/193736206-c1880d92-830f-4c49-bae3-b0e0b4763119.png">

Reply:
엇..!! 몰랐던 내용인데 감사합니다!! 2단계에서 적용하도록 하겠습니다😃